Sources/BraintreeCore/Analytics/BTAnalyticsService.swift

```diff
@@ -6,16 +6,27 @@ class BTAnalyticsService: Equatable {
 
     /// The HTTP client for communication with the analytics service endpoint. Exposed for testing.
     var http: BTHTTP?
-    
+
     /// The FPTI URL to post all analytic events.
     static let url = URL(string: "https://api-m.paypal.com")!
 
+    // MARK: - Private Properties
+
     private let apiClient: BTAPIClient
+    private let timerInterval: Double
+
+    private var timer: Timer?
+    private var events: [FPTIBatchData.Event] = []
 
     // MARK: - Initializer
 
-    init(apiClient: BTAPIClient) {
+    init(apiClient: BTAPIClient, timerInterval: Double = 30) {
```

Comment:
I ran through a bunch of different flows to find the "sweet spot" here. 30s seemed to be the most ideal for every flow except card, which tended to be faster. We could also consider bumping this down to 20s, but that seemed to be about as low as we would want to go while still getting a significant number of batched events. Anything below 20 (5, 10, 15) there were times we would just send a single event. Curious to hear what other folks think.